### PR TITLE
Add data currently associated with Culture and Work Environment to JobPosters

### LINKED ISSUE #1021: Backend - Update the database model and API for Work Environment and Culture
# Description
Update the database, eloquent models, and api for Work Environment and Culture, to be in line the new job builder #745.

# Acceptance Criteria
- [x] This data now needs to be stored per job.
- [x] It needs to remain backwards compatible (new and old jobs need to be viewable.
- [ ] There needs to be an api route for a manager to get their most recent settings
- [x] PUT requests need validation
- [x] api controllers need unit tests

### PR DESCRIPTION
**Pull request by tristan-orourke**
Resolves #1021.

### CODE REVIEW

File: database/migrations/2019_06_11_220005_convert_existing_work_env_and_culture_to_job_poster.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                    $workEnvTrans = DB::table('work_environment_translations')->where('work_environment_id', $workEnv->id)->where('locale', $locale)->first();
+                    if ($workEnvTrans) {
+                        DB::table('job_poster_translations')->updateOrInsert(
+                            ['job_poster_id' => $job->id, 'locale' => $locale],
+                            ['work_env_description' => $workEnvTrans->things_to_know]
+                        );
+                    }
+                }
+            }
+            $culture = DB::table('team_cultures')->where('manager_id', $job->manager_id)->first();
```
Comment (Khristophor) [inline database/migrations/2019_06_11_220005_convert_existing_work_env_and_culture_to_job_poster.php]: Just need to add a line in here somewhere to populate the `culture_summary` field (if applicable)

File: database/migrations/2019_06_10_220851_add_work_env_and_culture_to_job_poster.php [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+     */
+    public function up()
+    {
+        Schema::table('job_posters', function (Blueprint $table) {
+            $table->integer('team_size')->nullable();
+
+            // Work Env checkboxes
+            // Storing as a json because these features are likely to change frequently
+            // This way the frontend can store and check whatever tags it wants to.
+            $table->json('work_env_features')->nullable();
```
Comment (Khristophor) [inline database/migrations/2019_06_10_220851_add_work_env_and_culture_to_job_poster.php]: Should add the proposed shape of the json as a comment in the `JobPoster` model and delete the comments here.

File: database/migrations/2019_06_11_220005_convert_existing_work_env_and_culture_to_job_poster.php (hunk 1 above)
Comment (tristan-orourke) [reply database/migrations/2019_06_11_220005_convert_existing_work_env_and_culture_to_job_poster.php]: Fixed in d5f3d45ee726b00494c204a3f1dc72bcbb0e5e80

File: database/migrations/2019_06_10_220851_add_work_env_and_culture_to_job_poster.php (hunk 2 above)
Comment (tristan-orourke) [reply database/migrations/2019_06_10_220851_add_work_env_and_culture_to_job_poster.php]: Addressed in d5f3d45ee726b00494c204a3f1dc72bcbb0e5e80

**Pull request merged**